src/main/java/htsjdk/samtools/reference/FastaReferenceWriterBuilder.java

```diff
@@ -127,9 +155,18 @@ public FastaReferenceWriterBuilder setBasesPerLine(final int basesPerLine) {
     /**
      * Set the output index file to write to. Doesn't (currently) support compressed filenames.
      */
-    public FastaReferenceWriterBuilder setIndexFile(final Path indexFile) {
-        this.indexFile = indexFile;
-        this.indexOutput = null;
+    public FastaReferenceWriterBuilder setIndexFile(final Path faiIndexFile) {
+        this.faiIndexFile = faiIndexFile;
+        this.faiIndexOutput = null;
+        return this;
+    }
+
+    /**
+     * Set the output index file to write to. Doesn't (currently) support compressed filenames.
+     */
+    public FastaReferenceWriterBuilder setGziIndexFile(final Path gziIndexFile) {
+        this.gziIndexFile = gziIndexFile;
+        this.gziIndexOutput = null;
```

Comment:
Shouldn't this set makeGziOutput = true for consistency ? Also, does the filename have to conform (".gzi") ? Will the index work if its something else ?

Reply:
Hmm... no a gzi index named anything other than fastafile.gzi will fail. I don't see this as a unique problem given that the user could set the fai index file to any arbitrary name elsewhere. My inclination is to simply leave it as is.

Reply:
Also we don't need to set makeGziOutput = true as the make<FILE>Output flags can be more accurately thought of as "should this tool automatically generate this output given that the user hasn't provided me with a path for this file explicitly"